ivy/data_classes/array/array.py

```diff
@@ -792,7 +792,7 @@ def __bool__(self):
         return self._data.__bool__()
 
     def __dlpack__(self, stream=None):
-        return self._data.__dlpack__()
+        return ivy.to_dlpack(self)
 
     def __dlpack_device__(self):
```

Comment:
have we tested if `__dlpack_device__` works correctly?

Reply:
No, as `tf` and `paddle` doesn't support these methods yet.